hazelcast/src/main/java/com/hazelcast/ringbuffer/Ringbuffer.java

```diff
@@ -340,6 +340,11 @@
      * If there are less items available than {@code minCount}, then this call
      * blocks.
      * <p>
+     * <b>Warning:</b>
+     * <p>
+     * These blocking calls consume server memory and if there are many calls,
+     * it can be possible to see leaking memory or OOME.
```

Comment:
I guess you can include in this PR delaying preallocating `items` and `seqs` inside the `ReadResultSetImpl`. Probably we can leave `ReadManyOperation` as-is, but rather expand the `items` and `seqs` as new items are added, maybe even by coordinating with how many will be added exactly in `RingbufferContainer#readMany`, instead of having some power-of-two expansion like `ArrayList` or wrapper objects like `LinkedList`.

Reply:
done